cpp/include/cudf/lists/list_device_view.cuh

```diff
@@ -190,4 +190,19 @@ class list_device_view {
   };
 };
 
+/**
+ * @brief returns size of the list by row index
+ *
+ */
+struct list_size_functor {
+  column_device_view const d_column;
```

Comment:
Seems like this should have a constructor that verifies the column is a list column.

Reply:
It's used in host and also device codes. `#if defined(__CUDA_ARCH__)` is added to verify this.

Reply:
added constructor.